src/Workspaces/Core/Portable/Storage/AbstractPersistentStorageService.cs

```diff
@@ -203,39 +195,6 @@ private ReferenceCountedDisposable<IChecksummedPersistentStorage> TryCreatePersi
             }
         }
 
-        private void LocationServiceStorageLocationChanging(object sender, PersistentStorageLocationChangingEventArgs e)
-        {
-            ReferenceCountedDisposable<IChecksummedPersistentStorage> storage = null;
-
-            lock (_lock)
-            {
-                if (e.SolutionId != _currentPersistentStorageSolutionId)
-                {
-                    return;
-                }
-
-                // We will transfer ownership in a thread-safe way out so we can dispose outside the lock
-                storage = _currentPersistentStorage;
-                _currentPersistentStorage = null;
-                _currentPersistentStorageSolutionId = null;
-            }
-
-            if (storage != null)
-            {
-                if (e.MustUseNewStorageLocationImmediately)
-                {
-                    // Dispose storage outside of the lock. Note this only removes our reference count; clients who are still
-                    // using this will still be holding a reference count.
-                    storage.Dispose();
-                }
-                else
-                {
-                    // make it to shutdown asynchronously
-                    Task.Run(() => storage.Dispose());
-                }
```

Comment:
this code is terrifying.  the idea of just randomly spinning up work onto a random thread to dispose objects just seems awful.  I'm glad it is gone with this change.